### PR TITLE
Support patch releases for Konflux builds

### PR DESCRIPTION
**Pull request by pierDipi**
This will support pushing images to a different tag as soon as we bump the patch version

### CODE REVIEW

File: hack/generate/update-pipelines.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+readonly version=$(yq read olm-catalog/serverless-operator/project.yaml project.version)
+# shellcheck disable=SC2038
+find .tekton/ \( -name "*-push.yaml" \) | xargs -I{} yq write --inplace "{}" "spec.params(name==additional-tags).value[0]" "${version}"
```
Comment (creydr) [inline hack/generate/update-pipelines.sh:7]: Not really sure, but could we add a comment [here](https://github.com/openshift-knative/hack/blob/d63bd441112550420fc587cd4548252b215670df/pkg/prowgen/prowgen_konflux.go#L316), that the version should be on `Tags[0]`?
Just to reduce the probability that we break this in future

Comment (pierDipi) [reply hack/generate/update-pipelines.sh:7]: I will do

Comment (pierDipi) [reply hack/generate/update-pipelines.sh:7]: https://github.com/openshift-knative/hack/pull/339

**Comment by pierDipi** [comment]
Konflux pipelines still running after 1 day :fire:

**Comment by pierDipi** [comment]
https://github.com/openshift-knative/serverless-operator/pull/2936/checks?check_run_id=31492343304

**Pull request merged**